src/shared/vscode/commands2.ts

```diff
@@ -69,15 +69,28 @@ export class Commands {
                 throw new Error(`Command "${id}" cannot be re-registered`)
             }
 
-            return new CommandResource<Callback>({ id, factory: throwOnRegister }, this.commands)
+            const info = {
+                id: id,
+            }
+            return new CommandResource<Callback>({ info, factory: throwOnRegister }, this.commands)
         }
     }
 
     /**
      * Registers a new command with the VS Code API.
+     *
+     * @param info command id (string) or {@link CommandInfo} object
+     * @param callback command implementation
      */
-    public register<T extends Callback>(id: string, callback: T): RegisteredCommand<T> {
-        const resource = new CommandResource({ id, factory: () => callback }, this.commands)
+    public register<T extends Callback>(info: string | CommandInfo<T>, callback: T): RegisteredCommand<T> {
```

Comment:
@JadenSimon looking for a way to pass flags like `autoconnect` and (future) `notelemetry` to `runCommand()`. I added `autoconnect` field to `CommandInfo` and plumb that through all the codepaths.